test/tests/functional/pbs_test_tpp.py

```diff
@@ -0,0 +1,112 @@
+        j = Job(PBSROOT_USER)
+        if set_attrib:
+            j.set_attributes(set_attrib)
+        if interactive:
+            j.interactive_script = [('hostname', '.*'),
+                                    ('export PATH=$PATH:%s' %
+                                     self.exec_path, '.*'),
+                                    ('qstat', '.*')]
+        j.set_sleep_time(sleep)
+        jid = self.server.submit(j)
+        if exp_attrib:
+            self.server.expect(JOB, exp_attrib, id=jid)
+        return jid
+
+    def submit_resv(self, set_attrib=None, exp_attrib=None):
+        r = Reservation(PBSROOT_USER)
+        if set_attrib:
+            r.set_attributes(set_attrib)
+        rid = self.server.submit(r)
+        if exp_attrib:
+            self.server.expect(RESV, exp_attrib, id=rid)
+        return rid
+
+    def test_comm_with_mom(self):
+        """
+        Test the installation of communication daemon during PBS installation.
+        """
+        msg = "Need atleast 2 moms as input. use -pmoms=<m1>:<m2>"
+        if len(self.moms) < 2:
+            self.skip_test(reason=msg)
+        log_msgs = ["TPP initialization done",
+                    "Single pbs_comm configured, " +
+                    "TPP Fault tolerant mode disabled",
+                    "Connected to pbs_comm %s.*:17001" % self.server.shortname]
+        for msg in log_msgs:
+            self.server.log_match(msg, regexp=True)
+            self.scheduler.log_match(msg, regexp=True)
+            for mom in self.moms.values():
+                self.mom.log_match(msg, regexp=True)
+        server_ip = socket.gethostbyname(self.server.hostname)
+        msg = "Registering address %s:15001 to pbs_comm" % server_ip
+        self.server.log_match(msg)
+        msg = "Registering address %s:15004 to pbs_comm" % server_ip
+        self.scheduler.log_match(msg)
+        for mom in self.moms.values():
+            ip = socket.gethostbyname(mom.shortname)
+            msg1 = "Registering address %s:15003 to pbs_comm" % ip
+            msg2 = "Leaf registered address %s:15003" % ip
+            mom.log_match(msg1)
+            self.comm.log_match(msg2)
+        # Submit job
+        jid = self.submit_job(exp_attrib={'job_state': 'R'})
+        self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
+        self.server.log_match("%s;Exit_status=0" % jid)
+        # Submit multi-chunk job
+        set_attrib = {'Resource_List.select': '2:ncpus=1'}
+        jid = self.submit_job(set_attrib, exp_attrib={'job_state': 'R'})
+        self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
+        self.server.log_match("%s;Exit_status=0" % jid)
+        # Submit interactive job
+        set_attrib = {ATTR_inter: ''}
+        jid = self.submit_job(set_attrib, interactive=True)
+        # Submit reservation
+        set_attrib = {'Resource_List.select': '1:ncpus=1',
+                      'reserve_start': int(time.time() + 5),
```

Comment:
5 seconds could be short if you want to check for the confirmed state. on slow machines the reservation may start running before you check for the confirmed state.